pkg/collector/corechecks/cluster/orchestrator/orchestrator.go

```diff
@@ -340,21 +345,43 @@ func (o *OrchestratorCheck) processNodes(sender aggregator.Sender) {
 	}
 	groupID := atomic.AddInt32(&o.groupID, 1)
 
-	messages, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID)
+	nodesMessages, clusterModel, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID)
 	if err != nil {
 		o.Warnf("Unable to process node list: %s", err) //nolint:errcheck
 		return
 	}
+	sendNodesMetadata(sender, nodesList, nodesMessages, o.clusterID)
 
+	clusterMessage, clusterErr := extractClusterMessage(o.orchestratorConfig, o.clusterID, o.apiClient, groupID, clusterModel)
+	if clusterErr != nil {
+		_ = o.Warnf("Could not collect orchestrator cluster information: %s, will still send nodes information", err)
+		return
+	}
+	if clusterMessage != nil {
```

Comment:
is it possible that this can be `nil`? as we early exit on node errors

Reply:
Yes, we could skip the resource if it was in the cache. 
https://github.com/DataDog/datadog-agent/blob/459d4698507b05da92494661c4477cd4e0303ca1/pkg/collector/corechecks/cluster/orchestrator/orchestrator_processor.go#L385-L393
We could remove this, but then things have to move to other functions 🤔